Adjustment practice

Jiayue-He opened the pull request.

Commit:
complete revising app

Commit:
complete description file

Commit:
adding feedback

Commit:
Create docs

Commit:
Delete docs

Commit:
Delete server.R

Commit:
Delete ui.R

Commit:
add docs file

Commit:
Delete rsconnect/shinyapps.io/psu-eberly directory

Commit:
fix Error in execCallbacks(timeoutSecs, all, loop$id) :
  invoke_wrapped: throwing std::runtime_error

Commit:
Merge branch 'adjustment_practice' of github.com:EducationShinyAppTeam/Logistic_Regression into adjustment_practice

Commit:
Delete README.md

Commit:
fix the error, not sure,
Error in execCallbacks(timeoutSecs, all, loop$id) :
  invoke_wrapped: throwing std::runtime_error

Commit:
Merge branch 'adjustment_practice' of github.com:EducationShinyAppTeam/Logistic_Regression into adjustment_practice

Commit:
adding reference

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+
+library(dplyr)
+library(shinycssloaders)
+
```
Missing the BoastUtils package! When I tried running this code on my laptop, it says that the script function which is in BoastUtils was not found.

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+
+source("helpers.R") 
+
+ui <- dashboardPage(
```
According to the style guide, I believe this should be:

ui <- list(
     dashboardPage( ...

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+      menuItem("References", tabName = "references", icon = icon("leanpub"))
+    ),
+    
+    tags$div(
```
Here is the updated PSU button code from the style guide!

tags$div(
        class = "sidebar-logo",
        boastUtils::sidebarFooter()
      )

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+    tabItems(
+      #Adding pre-requisites page to remove background from instructions page
+      
+      tabItem(tabName="prereq",
```
The tab items are out of order. I think it should be Overview page, then prerequisites page

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+  ######### Could be combined but left separate so easily understood#####################
+  
+  dashboardBody(
+    tags$head(
```
Not quite sure what lines 61 - 113 are supposed to do so I tried commenting it out and the app's background color for each page remained white! Consider asking Dr. Neil or Mr. Carey if it is ok to completely delete these lines of code.

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+      #Adding pre-requisites page to remove background from instructions page
+      
+      tabItem(tabName="prereq",
+              h3(strong("Logistic Regression Analysis")),
```
Before this, consider adding h2(Prerequisite) because I think it is important that each page have a title.

shravanisamala replied:
Delete strong()

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+    input$goButton
+    df = isolate(commonDf())
+    mod <- glm (y~x, data=df, family = binomial)
+    hl <- hoslem.test(mod$y, fitted(mod))
```
For some reason, when I open the run the code everything works but then when I look in the console it says hoslem.test not found. I believe you talked about this with Dr. Neil during the meeting and I google this function and it said there is a package called ResourceSelection that has this function. However, it says that the version of that package is not available for my version of R.

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+ui <- dashboardPage(
+  skin = 'yellow',
+  dashboardHeader(title = "Logistic Regression",
+                  tags$li(class="dropdown",
```
Just a small thing, but make sure to add spaces around equal signs. In the beginning, I think we enabled a setting in R that would give us small blue warnings on the side when there were these small style issues present!

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+      ),
+      
+      tabItem(tabName = "instruction",
+              
```
Don't forget to add Logistic Regression as the h1 title. Also, I noticed that the lists are in h4(). I do not think you need this as it makes the font look too big. The style guide usually laid out lists as p( and the tags$li; I would also consider changing this in the prerequisites page as well.

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+              h3(strong("About:")),
+              h4("This app allows you to explore how different factors can affect the outcome of the Logistic Regression Model and Empirical Logit Plot."),
+              br(),
+              h3(strong("Instructions:")),
```
remove strong from all titles and the color as well

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+              h4(tags$li("If the cumulative total for your dice roll reaches 20 within 10 questions, YOU WIN!")),
+              br(),
+              div(style = "text-align: center",
+                  bsButton(inputId = "go", label =  "Explore", icon("bolt"), style= "danger", size= "large", class='circle grow')
```
Another small thing, but I would indent each thing inside bsButton such as 
inputId = 
label = 
etc. 

Just for code readability purposes

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+      
+      
+      tabItem(tabName = "explore",
+              
```
Add a descriptive title in h2()

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+                        selected="deviance"),
+                      
+                      br(),
+                      actionButton("goButton", "New Sample", icon("paper-plane"),
```
https://educationshinyappteam.github.io/Style_Guide/htmlCSS.html

I was looking in the style guide (sorry I keep saying style guide I know I am mentioning it a lot!), and I believe there are certain colors you can use for buttons and I would just write out the color if it is one of those option instead of #ffff so future contributors will know what color it is more easily. Also I did this on one of the apps I was reviewing as well, but Dr. Neil said to take out the class = "circle grow".

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+      menuItem("Prerequisites", tabName = "prereq", icon = icon("book")),
+      menuItem("Overview",tabName = "instruction", icon = icon("dashboard")),
+      menuItem("Explore",tabName = "explore", icon = icon("wpexplorer")),
+      menuItem("Game", tabName = "qqq", icon = icon("gamepad")),
```
tabName could probably be game instead of qqq? I do not know if "qqq" stands for anything important, so if it does do not change it.

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+        tabName = "references",
+        withMathJax(),
+        h2("References"),
+        p(
```
Make sure you cite all of the packages you used!

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+    )
+  })
+  observeEvent(input$go,{
+    updateTabItems(session,"pages","explore")
```
change these to 
updateTabItems(
     session = session, 
     inputId = "pages", 
     selected = "explore")

shravanisamala replied:
And for the next five as well!

shravanisamala replied:
Also, when you switch the order of the tabs, these buttons will have to modified! Also, Dr. Pearl once told me that some of my buttons were too wordy when they had, for example, "Go to Prerequisites". He suggested just have the word "Go!" on it but changing the symbol to the page the button would go to. So for example, when you will have the go button on your overview page going to the prerequisite page, it would just say, "GO!" with the book icon next to it!

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+  
+  ############################Gray out buttons###############################
+  # observeEvent(input$start, {
+  #   updateButton(session, "answer", disabled = TRUE)
```
same as comment above but, make sure you make each attribute (not sure this is the right word) explicit. Always type out session = session and inputId = "" etc.

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+          "Carey, R. (2019). boastUtils: BOAST Utilities, R Package.
+                             Available from https://github.com/EducationShinyAppTeam/boastUtils"
+        )
+      ) #end of tabItem
```
Don't forget the copyright info at the end of the references page!

        br(),
        br(),
        br(),
        boastUtils::copyrightInfo()

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+          
+          
+        }
+      }
```
Check bracket alignment

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+
+  observeEvent(input$restart,{
+    updateButton(session, "submit", disabled = FALSE)
+    updateButton(session,"restart",disable =FALSE)
```
disabled not disable

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+    }
+    else{
+      if(length(index_list$list) == 1){
+        updateButton(session, "nextq", disabled = TRUE)
```
session = session etc.

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+    
+  })
+  
+  renderIcon()
```
I tried commenting this out and the app works without it. Was this something you were working on or can it be deleted?

shravanisamala reviewed (dismissed):
Hi Jiayue-He! Good job changing the ui.R. and server.R files to one app.R file! Just a few general things to consider: 
- I added many comments regarding the style because I was comparing the app side-by-side to the style guide. Make sure you go back in and add any necessary titles and remove and unnecessary details like the colons, strong(), and h4()s for bullet points. 
- Be sure to go through the whole code and add white space around = signs and other symbols where needed. 
- I would ask Dr. Neil and/or Mr. Carey about the huge chunks of code that are commented out and see if you can delete them. 
- Change the order of the tabs and fix the go buttons
- Cite the packages you haven't cited yet. Check if all of them are used with the stale packages function we talked about today
- Make sure your code is explicit everywhere and by this I mean the whole session = session, inputId = "" and so on 
- And for the overall code, I would try to go back when you have free time and fix the alignment and indenting of the code. I noticed that sometimes each line of text underneath the previous line was not consistently indented. For example, some lines are indented twice or thrice while some of the later lines are indented once. I would change them all to one just so we do not go past that 80 character guideline. 

Overall, great job! You did a lot of great work to get the app to where it is now! I hope some of my comments will help!

lbednarczyk commented on DESCRIPTION:
```diff
@@ -0,0 +1,23 @@
+  )
+DisplayMode: Normal
+URL: https://psu-eberly.shinyapps.io/Sample_App
+BugReports: https://github.com/EducationShinyAppTeam/Sample_App/issues
```
The website for URL and BugReports should be updated to say Logistic_Regression instead of Sample_App.

lbednarczyk commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+
+library(RColorBrewer) 
+#library(car)
+#library(rgdal)
```
You could delete the lines that have unused packages instead of leaving them commented out.

lbednarczyk commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+                                                where data is divided into recommended 10 groups. The p-value can 
+                                                determine the significance of the result.")),
+              br(),
+              h4(tags$li("Hosmer-Lemeshow Test Statstics")),
```
There is a typo in Statistics here.

lbednarczyk commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+                                                                            \\over exp_{ij}}$$')),
+              br(),
+              br(),
+              div(style = "text-align: center",bsButton("start","Go to the overview",
```
After you update the order of tabs to be first overview and then prerequisites, the label on this button should be changed to Explore.

lbednarczyk commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+              br(),
+              div(style = "text-align: center",bsButton("start","Go to the overview",
+                                                        icon("bolt"),style = "danger",
+                                                        size = "large",class="circle grow"))
```
None of the buttons should have the class = circle grow argument.

lbednarczyk commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+                      h3(strong(id='title', "Hosmer and Lemeshow goodness of fit test")),
+                      #br(),
+                      tableOutput("lemeshowDF"),
+                      tableOutput("obsexpDF"),
```
I was getting errors on the page where these outputs were supposed to be shown, the packages ResourceSelection and data.table should be added at the top of the app so that these outputs will run properly.

lbednarczyk commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+                  ),
+                  #set continue button
+                  div(style = "text-align: center",
+                      bsButton(inputId = "go1", label =  "Play the game!", icon("bolt"), style= "danger", size= "large", class='circle grow')
```
You could change the label on this button to just say "Play!".

lbednarczyk commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+                      ####variable options for 'MedGPA' dataset
+                      conditionalPanel(
+                        condition = "input.datatable == 'MedGPA'",
+                        selectInput(inputId="MedYvar", label="Select Response Variable Y",
+                                    choices = c("Acceptance"),
+                                    selected = 'Acceptance'),
```
Not sure if this selection is needed since there is only one option for the response variable for each of the possible data sets.

lbednarczyk commented on app.R:
```diff
@@ -0,0 +1,1215 @@
+                      br()
+                      
+                    ),
+                    mainPanel(
+                      plotOutput("empericalLogitPlot", width = "100%")%>% withSpinner(color="#ffa500")
```
This plot wouldn't load until I added the Stat2Data package at the beginning of the code.

lbednarczyk reviewed (commented):
Good job overall! Other than the comments that I left throughout the code, I would just make sure to go through the code and fix spacing issues and make all of the arguments explicit like Shravani said. Also make sure to add the missing packages so that everything runs smoothly.

lbednarczyk reviewed (dismissed):
Good job overall! Other than the comments that I left throughout the code, I would just make sure to go through the code and fix spacing issues and make all of the arguments explicit like Shravani said. Also make sure to add the missing packages so that everything runs smoothly.

Commit:
fix "could not find hoslem.test() function"

Commit:
fix "could not find setDT() function"

rpc5102 committed:
Cleaned up unused packages and applied styler formatting

rpc5102 committed:
Updated .gitignore

rpc5102 commented:
6aaa28de5649b9d7cb9eef64057b7bed1256bd2a was deployed to shinyapps.io to resolve an SSL Expiry error although I suggest further work to clean up.

Maxinewsu reviewed (dismissed):
**Recommended Priority: Urgent**
- Issues are all addressed.
- The link of URL and Bugreports on DESCRIPTION is not correct
- Missing screenshot on README.
- Add a citation on the overview page.
- Add a line "This app was last updated by xxx" at the bottom of the overview page.
- Switch the order of Prerequisite and Overview pages.
- The last three referenes on Reference page are not cited using the required format.
- The font size is too big compared to other shiny apps.

Maxinewsu committed:
updated

NurulSyafiqah-pixel reviewed (changes requested):
Based on the current style guide, there are few changes need to be made:

- On the overview page, 
1) the PennState logo should be deleted.
2) the 'About' should be replaced to the formal app title. *e.g. h1("Logistic Regression Analysis")

- On the Prerequisite page:
1) the title 'Logistic Regression Analysis' should be replaced to simply 'Prerequisite'.

- Heading issues:
1) On the app.R, h1() is used more than once. It should be just ONCE that is for the title on the overview page. 
2) Change other headlines that used h1() to their corresponding level (refer to Style Guide chapter 7: 7.1.4 Headers)

- also, the readme file should be updated.

Maxinewsu committed:
update

Maxinewsu committed:
update screenshot

NurulSyafiqah-pixel reviewed (dismissed):
All the changes addressed have been settled. Good to go.

Maxinewsu commented:
@sv101 Just a reminder that please review this app. Many thanks!

sv101 commented on app.R:
```diff
@@ -0,0 +1,1091 @@
+  dashboardHeader(
+    title = "Logistic Regression",
+    tags$li(
+      class = "dropdown",
```
The info button doesn't work. It seems to cause the app to close.

Maxinewsu committed:
fix info button

Maxinewsu commented:
@sv101 @NurulSyafiqah-pixel Hello, just a reminder that please re-review this app since I have made some changes based on your previous comments. Thanks!

Maxinewsu commented:
@sv101 Could you please help review this app? This was in the app moidified in the summer and I just realized that I need one more approval, thank you!

Sburke26 reviewed (changes requested):
Here are a few issues I noticed while reviewing the app:

The description lifecycle badge does not match the badge in the README.
The year in the link of the year badge is one year behind the one in brackets.
The app description is missing the proper heading in the README.
There are small grammatical errors. (Ex. Missing Periods).
The Citation part of the app is missing in the overview page.
Some of the button inputId's could be improved to be more descriptive and representative of their purpose.
Some of the code is messy/missing proper indentation and does not fit the proper style.
The Game tabname was changed to "qqq". I'm not sure if that is the best way to represent the tab.
When a question is incorrect, the first sentence: "Don't give up, try again !" is probably not the best here. After getting a question wrong you can't resubmit or "try again". You can either go to the next question or restart the game. Also, there is a space between the last word and the exclamation point.
There are a couple of messages in the console: "Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0. Please use `linewidth` instead." and "`geom_smooth()` using formula = 'y ~ x'".

tarynmch reviewed (changes requested):
Good app, I just noticed a few things. Not all the packages are listed in the references, there is no app description header in the README file, chapter title should be "Categorical Data" rather than "Chapter 10" in the description file, the i icon is more of an overview rather than a instructions, there is no cite as on the overview, and looking at the WAVE report has some instances of small text. Then in the game if you get a question wrong it says "try again" even though you can not try again. I suggest just saying incorrect or implementing a try again option. Within the coding, there are some unnecessary blank lines, on line 996 it has a warning triangle saying that 'disable' should be 'disabled', and I also noticed that the game tab inputID is 'qqq' which is not very descriptive of the purpose.

Sburke26 committed:
Update README.md

Sburke26 committed:
Update app.R

Updated Wording.

Updated some object names to better represent their purposes.

Updated the Overview Page.

Fixed Some of the code layout.

removed extra info button and fixed current one,

Sburke26 committed:
removed unused files

remove unused files from the www folder

Sburke26 committed:
Create ToDoList.txt

Added  a to do list to keep track of current and new issues that need to be taken care of.

Sburke26 committed:
Update ToDoList.txt

Sburke26 committed:
Update app.R

Fixed console message issues.

Added References.

Sburke26 committed:
Update ToDoList.txt

Sburke26 committed:
Delete Unecessary files

Sburke26 committed:
Create ScrapIdeas.R

Store code not currently used in the app, but may be used in the future.

Sburke26 committed:
Update app.R

Sburke26 committed:
Update ToDoList.txt

Sburke26 committed:
Update ToDoList.txt

"Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred" may not be fixable the way the app is made.

Sburke26 committed:
Update app.R

Sburke26 committed:
Update app.R

Sburke26 committed:
Add to Acknowledgements

tarynmch reviewed (changes requested):
Good work with updating the app. The UI looks good, but in the server, I noticed that some of the IDs are not camelcase (like gamescore), some things are not individually indented (line 749 and 755), the chapter name should be changed, and there is no alt text for some of the plots and images in the game. The only thing I suggest is to go through the question bank excel and make the capitalization consistent for the choices.
I also noticed that when using the Single Logistic Regression sliders, it said in the console, "glm.fit: fitted probabilities numerically 0 or 1 occurred."

Sburke26 committed:
Post-Review updates

Made fixes to code including the addition of alt texts, naming of objects, indentation, and deletion of unused code.

Made updates to the question bank file.

Removed unused images.

Fixed Chapter in description

Sburke26 commented:
Thank you for the review.
I made additional fixes to the app.
The "glm.fit: fitted probabilities numerically 0 or 1 occurred." message can't be removed due to the current way the data is simulated. 
I think I added alt text to everything but the logistic plot which uses renderPlotly() instead of renderPlot(). 
Please let me know if you notice anything else.

tarynmch reviewed (changes requested):
Looks good. The main thing I noticed was that when you click game it says, "Select your answer from below" but after you click next it says, "pick an answer from below." I would just make sure that is concise. Then make sure randNum is camelcase. I also suggest that for the game, if you want to get to 20 points within 10 questions, you could add a question counter. Otherwise, the app looks great!

Sburke26 committed:
Update app.R

Sburke26 committed:
Update app.R

Sburke26 commented:
Thank you for the review. I made some changes and added a question counter. Please let me know if you see anything else.

Sburke26 committed:
minor updates

Sburke26 committed:
Update Logistic Plot

Sburke26 committed:
Update app.R

Sburke26 committed:
Update app.R

tarynmch reviewed (dismissed):
Looks good! The only thing I suggest is that on the Overview page in the instructions, be sure to mention that the New Sample button is within the Explore tab. Otherwise, be sure to get rid of the unnecessary code in the app.R before you merge the branches. Great work.

Sburke26 committed:
updates

Sburke26 committed:
Update app.R

DavisJiwooIm committed:
Update app.R

neilhatfield committed:
Update app.R

Remove an problematic install.packages command

neilhatfield committed:
First set of final updates

neilhatfield committed:
Clean Up Explore Page

Cleaning up the explore page to improve efficiency and table accessibility

neilhatfield committed:
Remove Unnecessary Packages and Commands

Remove unnecessary packages and code from the app; deleted helpers file that contained elements counter to the Style Guide.

neilhatfield committed:
Update app.R

Clean up of alt text

neilhatfield committed:
Update app.R

Reorganizing Game Server code

neilhatfield committed:
Reformatting Game

Initial work towards reformatting the game page of the app
